tests/context.rs

```diff
@@ -0,0 +1,192 @@
+#![cfg(test)]
```

Comment:
These are unit tests! Move them into src/context.rs in a mod tests {}. tests/ is for integration tests.